CRM/Contact/Form/Search.php

```diff
@@ -822,6 +814,13 @@ public function postProcess() {
       return;
     }
     else {
+      if (array_key_exists($this->_searchButtonName, $_POST) ||
+        ($this->_force && !$crmPID)
+      ) {
+        //reset the cache table for new search
+        $cacheKey = "civicrm search {$this->controller->_key}";
+        Civi::service('prevnext')->deleteItem(NULL, $cacheKey);
+      }
```

Comment:
I don't understand what's being done here.  I'm not saying it's wrong; it's almost certainly right.  Just that I don't understand.

Reply:
@MegaphoneJon i think that is covered in a separate PR i think see https://github.com/civicrm/civicrm-core/pull/14655